### PR TITLE
Hot seller section tests

### PR DESCRIPTION
**Pull request by RaduMoldovan30**
**Six items displayed at hot sellers**
GIVEN you are on Home page
THEN 6 products are displayed at hot sellers section

**Item redirects correctly on home page**
GIVEN you are on Home page
WHEN you select any of the 6 items
THEN you are redirected to a new page
AND at least one value displayed is matching the initial selected item

### CODE REVIEW

File: tests/raduMld.spec.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  const pageTitle = await page.locator(".product-info-main .base").innerText();
+  let matchFound = false;
+  for (let i = 0; i < pageTitles.length; i++) {
+    if (pageTitle === pageTitles[i]) {
+      console.log(`The home title name ${pageTitles[i]} is the same as product title name ${pageTitle}`);
+      matchFound = true;
+      break;
+    }
+  }
```
Comment (adrianmaciuc) [inline tests/raduMld.spec.js:33]: I Don t think this is needed

File: tests/raduMld.spec.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  const randomProduct =
+  numberOfProducts[Math.floor(Math.random() * numberOfProducts.length)];
+  const pageTitles = await page.locator(".product-item-link").allInnerTexts();
+  await randomProduct.click();
+
+  const pageTitle = await page.locator(".product-info-main .base").innerText();
+  let matchFound = false;
+  for (let i = 0; i < pageTitles.length; i++) {
+    if (pageTitle === pageTitles[i]) {
+      console.log(`The home title name ${pageTitles[i]} is the same as product title name ${pageTitle}`);
```
[the remark this answers is not in the document]
Comment (RaduMoldovan30) [reply tests/raduMld.spec.js:29]: I remember that in Test Complet at Smedix we used console.log in tests, to make sure each time a test was run we had logs to look at, and if the test failed to know where and why it did not pass.

[the remark this answers is not in the document]
Comment (adrianmaciuc) [reply tests/raduMld.spec.js]: Time to step up your game. Be more professional

File: tests/raduMld.spec.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  const numberOfProducts = await page.locator(".product-item").count();
+  expect(numberOfProducts).toBe(6);
+});
+
+
+test("Select random products from home page", async ({ page }) => {
+  await page.goto("https://magento.softwaretestingboard.com/");
+  const numberOfProducts = await page.locator(".product-item").all();
+  const randomProduct = numberOfProducts[Math.floor(Math.random() * numberOfProducts.length)];
+  const pageTitles = await page.locator(".product-item-link").allInnerTexts();
```
Comment (pmirel) [inline tests/raduMld.spec.js:21]: I like this one

**Review by pmirel — APPROVED** [review]

File: tests/raduMld.spec.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  await randomProduct.click();
+  const pageTitle = await page.locator(".product-info-main .base").innerText();
+  expect(pageTitles).toContain(pageTitle);
```
Comment (adrianmaciuc) [inline tests/raduMld.spec.js:24]: you forgot an await here

File: tests/raduMld.spec.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+test("Check if the user is on the Home Page", async ({ page }) => {
+  await page.goto("https://magento.softwaretestingboard.com/");
+  await expect(page).toHaveURL("https://magento.softwaretestingboard.com/");
+  await expect(page).toHaveTitle("Home Page");
+});
+
+test("Count products on the home page", async ({ page }) => {
+  await page.goto("https://magento.softwaretestingboard.com/");
+  const numberOfProducts = await page.locator(".product-item").count();
+  expect(numberOfProducts).toBe(6);
```
Comment (adrianmaciuc) [inline tests/raduMld.spec.js:13]: you forgot an await here

**Review by pmirel — APPROVED** [review]
clean and simple

**Review by adrianmaciuc — APPROVED** [review]

**Pull request merged**